src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_STORAGE_SIZE = 10;
+
+    private final K[] keys;
+    private final V[] values;
+    private int currentSize;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_STORAGE_SIZE];
+        values = (V[]) new Object[MAX_STORAGE_SIZE];
+        currentSize = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndexByKey(key);
+
+        if (index != -1) {
+            values[index] = value;
+        } else if (currentSize < MAX_STORAGE_SIZE) {
+            keys[currentSize] = key;
+            values[currentSize] = value;
+            currentSize++;
+        }
     }
 
     @Override
     public V get(K key) {
+        int index = findIndexByKey(key);
+        if (index != -1) {
+            return values[index];
+        }
         return null;
     }
 
     @Override
     public int size() {
+        return currentSize;
+    }
+
+    private int findIndexByKey(K key) {
+        for (int i = 0; i < currentSize; i++) {
+            if ((key == null && keys[i] == null) || (key != null && key.equals(keys[i]))) {
```

Comment:
```suggestion
            if (key == null && keys[i] == null || key != null && key.equals(keys[i])) {
```
Redundant brackets. Read about logical operator priority

Reply:
Thank you very much for the explanation